fish-rust/src/common.rs

```diff
@@ -1642,55 +1649,61 @@
     // overhead since there's no actual race condition here - even if multiple threads call this
     // routine simultaneously the first time around, we just end up needlessly querying uname(2) one
     // more time.
-    *IS_WINDOWS_SUBSYSTEM_FOR_LINUX
-}
-
-fn slice_contains_slice<T: Eq>(a: &[T], b: &[T]) -> bool {
-    a.windows(b.len()).any(|aw| aw == b)
-}
-
-#[cfg(not(windows))]
-static IS_WINDOWS_SUBSYSTEM_FOR_LINUX: Lazy<bool> = Lazy::new(|| false);
-#[cfg(windows)]
```

Comment:
my bad, I somehow mixed up cfg(windows) with the WSL define. 
LGTM!

FWIW could keep using macros like that with `fish-rust/src/compat.{c,rs}` but it's probably not worth it here